File: AoC2023-GO/OneTwo/onetwo.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	file, err := os.Open("../../Data2023/DayOneData.txt")
	if err != nil {
		fmt.Print(err)
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)

	numbers := []string{"1", "2", "3", "4", "5", "6", "7", "8", "9", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"}

	maxValue := 0
	for scanner.Scan() {
		firstLowestIndex := len(scanner.Text())
		fistNumber := 0
		lastHighestIndex := -1
		lastNumber := 0

		for i := 1; i < len(numbers)+1; i++ {
			number := numbers[i-1]
			firstIndex := strings.Index(scanner.Text(), number)
			lastIndex := strings.LastIndex(scanner.Text(), number)

			if firstIndex == -1 || lastIndex == -1 {
				continue
			}

			if firstIndex < firstLowestIndex {
				firstLowestIndex = firstIndex
				if i > 9 {
					i -= 9
				}
				fistNumber = i
			}

			if lastIndex > lastHighestIndex {
				lastHighestIndex = lastIndex
				if i > 9 {
					i -= 9
				}
				lastNumber = i
			}
		}

		maxValue += fistNumber*10 + lastNumber
	}

	if err := scanner.Err(); err != nil {
		fmt.Print(err)
	}

	fmt.Println(maxValue)
}

File: AoC2023/AoC2023/One/OneTwo.cs
namespace AoC2023.One;

public static class OneTwo
{
    public static int Run(string dataFilepath)
    {
        var numbers = new[] {"1", "2", "3", "4", "5", "6", "7", "8", "9", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};
        
        var maxValue = 0;
        foreach (var dataRow in File.ReadLines(dataFilepath))
        {
            var firstLowestIndex = dataRow.Length;
            var fistNumber = 0;
            var lastHighestIndex = -1;
            var lastNumber = 0;
            
            for(var i = 1; i < numbers.Length + 1; i++)
            {
                var number = numbers[i - 1];
                var firstIndex = dataRow.IndexOf(number, StringComparison.OrdinalIgnoreCase);
                var lastIndex = dataRow.LastIndexOf(number, StringComparison.OrdinalIgnoreCase);

                if (firstIndex == -1 || lastIndex == -1)
                    continue;
                
                if (firstIndex < firstLowestIndex)
                {
                    firstLowestIndex = firstIndex;
                    if (i > 9)
                        i -= 9;
                    fistNumber = i;
                }

                if (lastIndex > lastHighestIndex)
                {
                    lastHighestIndex = lastIndex;
                    if (i > 9)
                        i -= 9;
                    lastNumber = i;
                }
            }
            
            maxValue += fistNumber * 10 + lastNumber;
        }

        return maxValue;
    }
}